### PR TITLE
Added comparison_helper that uses python eval to check value comaprisons

### LINKED ISSUE #146: Warnings come up for loan terms that are well within boundaries (W0182)
With values like '715' we get warnings like "when present, 'loan term' should be less than 1200(100 years)."

### PR DESCRIPTION
**Pull request by jcadam14**
Closes #146 

Turns out our is_less_than, is_greater_than and is_greater_than_or_equals all had the same issue.  So I created a comparison_helper function that uses python eval() to convert the value to its type (so we're not beholden to int, float, etc in the functions) and then uses eval again with the passed in comparison sign to evaluate the value and the limit.

Also beefed up the pytests for all three functions to use values that would actually cause the error.  Simple strings like "1" and "2" won't cut it.

### CODE REVIEW

File: src/regtech_data_validator/check_functions.py [hunk 1]
```diff
@@ -39,6 +39,21 @@ def _check_blank_(value: str, check_result: bool, accept_blank: bool = False) ->
...
+    Then, if actual value exists, uses python eval() to convert string to numerical values
+    and then evals again to use the passed in sign to compare the value to the limit.  This
+    ensures accurate comparison checks.
+    """
+    if accept_blank and not value.strip():
+        return True
+    elif not accept_blank and not value.strip():
+        return False
+    else:
+        return (bool)(eval(f"eval(value) {sign} eval(limit)"))
```
Comment (lchen-2101) [inline src/regtech_data_validator/check_functions.py:54]: not a fan of using evals in general, why can't we just cast them to int or float as I believe eval would run into the same issues with non-numerical strings?
also, I think having this just be the blank check would be enough, and have the actual comparison logic in the original functions that are named as the logic.
if we want to shortcut the comparison, then we can just have this blank_checker return True, False, or None, and if it's None, then do the logical comparison.

**Review by hkeeler — DISMISSED** [review]
This is pretty slick. My only concern is that `eval()` is the type of thing security code scanning tools yell at you for...mostly rightfully so. In our case, this data is completely managed by our own config file, so I'm not concerned about an actual code injection type issue, but we may have to justify why we're doing this if this pops up on a scan report.

**Comment by hkeeler** [comment]
Actually, scratch what I said. We _don't_ control what the `value` part of the `eval`, that comes in through the file. Duh. And yes, we do have earlier syntactical checks that make sure the values are `int`s or `float`s, so theoretically we'd never hit a scenario where someone could pass in python code in the file to be evaluated, but I think that's a risk that we probably don't want to have to think about or try to justify.

File: src/regtech_data_validator/check_functions.py (hunk 1 above)
Comment (hkeeler) [reply src/regtech_data_validator/check_functions.py:54]: The original intent for the phase 2 validations was to convert all the data to be typed appropriately, and then have all the phase 2 check functions be typed as well. Then you wouldn't have to redo all the string parsing logic/handling over again that the phase 1 validations already did.

That work obviously never happed, but I still like the idea. I know it doesn't help here, but  something we should evaluate post-MVP.

Comment (jcadam14) [reply src/regtech_data_validator/check_functions.py:54]: > not a fan of using evals in general, why can't we just cast them to int or float as I believe eval would run into the same issues with non-numerical strings? also, I think having this just be the blank check would be enough, and have the actual comparison logic in the original functions that are named as the logic. if we want to shortcut the comparison, then we can just have this blank_checker return True, False, or None, and if it's None, then do the logical comparison.

Was using eval for future cases where we might have ints or floats, and if we do a conversion to either, all three functions would need to implement the same code so I put it in one place.   Each would need to check for existence, then the specific type, then convert and then do the comparison.  The existing blank checker checked for existence afterwards and with casting, we have to check all that before.

**Comment by jcadam14** [comment]
> Actually, scratch what I said. We _don't_ control what the `value` part of the `eval`, that comes in through the file. Duh. And yes, we do have earlier syntactical checks that make sure the values are `int`s or `float`s, so theoretically we'd never hit a scenario where someone could pass in python code in the file to be evaluated, but I think that's a risk that we probably don't want to have to think about or try to justify.

Ok updated to use the same idea, but scrapped eval, convert to float, and pass in an operator function instead.

**Review by lchen-2101 — APPROVED** [review]
LGTM

**Pull request merged**